### ISSUE TITLE
[BUG] Cardinality error for missing $lang

### ISSUE THREAD

**Issue by lguariento** [eXist-db/AtomicWiki#84]
[This post](https://exist-db.org/exist/apps/wiki/blogs/eXist/XQuery31) is not shown and in its place there's a nice cardinality error. Does this happen because the language is missing so that [this function](https://github.com/eXist-db/AtomicWiki/blob/272ac0d724a37913550cd44b296f4cd662e237ca/src/main/xar-resources/modules/atomic.xql#L13) complains?

If that's the case, shall we introduce a condition?

**Comment by lguariento** [comment]
PS: also, a curiosity. Why does the error message say `[at line 309, column 36, source: /db/apps/wiki/modules/atomic.xql]` when there's not even a line 309 in `atomic.xql` (the function signature mentioned in the error message is way up)?

Even more curiously, there's some relevant code at, indeed, line 309, although of a different file: `store.xql`, [here](https://github.com/eXist-db/AtomicWiki/blob/faa445547ec4a448bc4afe3314ee9d42d029e383/src/main/xar-resources/modules/store.xql#L309). Is this expected behaviour/error messaging?

**Comment by lguariento** [comment]
This is connected I suppose: https://exist-db.org/exist/apps/wiki/blogs/eXist/?start=46. It happens between 44 and 51.

**Comment by lguariento** [comment]
Also, I've found a blog without `id`.

```
xquery version "3.1";

declare namespace output="http://www.w3.org/2010/xslt-xquery-serialization";
declare namespace wiki="http://exist-db.org/xquery/wiki";
declare namespace atom="http://www.w3.org/2005/Atom";
declare namespace xhtml="http://www.w3.org/1999/xhtml";
declare namespace http="http://expath.org/ns/http-client";

declare option output:method "html5";
declare option output:media-type "text/html";

declare variable $local:WIKI_ROOT := "http://exist-db.org/exist/apps/wiki";

response:set-header("Cache-Control", "no-cache"),

let $start := 13
let $count := 1
let $timelineURLString := "https://exist-db.org/exist/apps/wiki/modules/feeds.xql?feed=eXist&amp;start=" || $start || "&amp;count=" || $count
let $data := http:send-request(
    <http:request method='get' />,$timelineURLString)
  let $entries :=
    for $entry in $data//atom:entry
    let $date := ($entry/atom:updated, $entry/atom:published)[1]
    return $entry//wiki:id
    
    return $entries

```

Returns `<wiki:id xmlns:wiki="http://exist-db.org/xquery/wiki"/>`

It's the eXist-db 5.0.0 RC8. In fact, clicking on its link at the top of the page [here](http://exist-db.org/exist/apps/wiki/blogs/eXist/?start=13) doesn't do anything.

**Comment by lguariento** [comment]
PS: if this is about three separate issues I'll open two other issues.

**Comment by lguariento** [comment]
This might also be connected: searching for xquery (!!!) returns a cardinality error: https://exist-db.org/exist/apps/wiki/blogs/eXist/search.html?q=xquery . It doesn't get the entry `id` for building the URI I suppose.

**Comment by joewiz** [comment]
@lguariento Thanks for opening the issue about language cardinality issues. Further study is needed. Would you like to have a full copy of the app as it is on exist-db.org so you can explore further on your local system?

I fixed the 5.0.0 RC8 error on the server. The problem was an incorrect `@src` value in `<atom:content src=".md">` in the Atom entry for this article at `/db/apps/wiki/data/blogs/eXist/eXistdb500RC8.atom`. Changing it to `eXistdb500RC8.md` did the trick.

**Comment by lguariento** [comment]
Great. Yes, I'd like to have the full set so that I can check things on my own.

I find it ironical that searching for 'xquery' returns a nasty error :) .

**Comment by lguariento** [comment]
Also, @joewiz , does the issue I fixed the 5.0.0 RC8 you fixed need time to be replicated? I still get the error.

**Comment by lguariento** [comment]
This is the same issue discussed [here](https://github.com/eXist-db/AtomicWiki/issues/81). Closing this.
